src/extension.ts

```diff
@@ -435,43 +435,25 @@ function registerCommands(
   context.subscriptions.push(
     commands.registerCommand(
       "sourcery.coding_assistant",
-      ({
-        view,
-        request,
-        message,
-      }: {
-        view: string;
-        request: string;
-        message?: ServerRequest;
-      }) => {
-        vscode.commands.executeCommand("sourcery.chat.focus").then(() => {
-          // Use the editor selection unless a range was passed through in
-          // the message
-          let selectionLocation = getSelectionLocation();
-          if (message?.context_range) {
-            selectionLocation = {
-              uri: selectionLocation.uri,
-              range: message.context_range,
-            };
-          }
-          let { activeFile, allFiles } = activeFiles();
-
-          let params: ExecuteCommandParams = {
-            command: "sourcery.coding_assistant",
-            arguments: [
-              {
-                view,
-                request,
-                message,
-                selected: selectionLocation,
-                active_file: activeFile,
-                all_open_files: allFiles,
-              },
-            ],
-          };
-          console.log(params);
-          languageClient.sendRequest(ExecuteCommandRequest.type, params);
-        });
+      (request: { selected?: { uri: string; range: vscode.Range } }) => {
+        // Note: request has other keys - we're only declaring the ones that are relevant to this function.
+        // It's not necessary or useful to provide additional keys because this function is never called
+        // explicitly, so there's never any type checking.
+        let selectionLocation = getSelectionLocation();
+        let { activeFile, allFiles } = activeFiles();
+
+        let params: ExecuteCommandParams = {
+          command: "sourcery.coding_assistant",
+          arguments: [
+            {
+              active_file: activeFile,
+              all_open_files: allFiles,
+              ...request,
+              selected: { ...selectionLocation, ...request.selected }, // request may override selected
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** ‏I think merging these request arguments and IDE values here is asking for trouble

There's a few possible downsides:
- When working with this data on the backend it's not obvious where each piece of information has come from. Is it the request, is it the IDE, or is it some combination of the two?
- The request may accidentally replace `active_file`
- This logic would need to be replicated in JetBrains

I think this would be much clearer if it was separated into it's constituent parts, both here and in the backend. e.g.
```js
       let params: ExecuteCommandParams = {
          command: "sourcery.coding_assistant",
          arguments: [
            request,
            {
              active_file: activeFile,
              all_open_files: allFiles,
              selected: selectionLocation,
            }
          ]
        };
```

If there are overrides in the request this can be handled explicitly in the backend

I know this is a variation of the flat vs nested comment thread but I am very uncomfortable with it

Reply:
Or alternatively
```js
       let params: ExecuteCommandParams = {
          command: "sourcery.coding_assistant",
          arguments: [
            { 
              request: request,
              ide: {  // or some better name
                active_file: activeFile,
                all_open_files: allFiles,
                selected: selectionLocation,
            }
          ]
        };
```

Reply:
I'm more in favour of a single object, and I take your point about not knowing where some bits of information may have come from. One of the reasons I'm super reluctant to do this is to avoid the kind of silly `{"message": {"request": {"data": ...}}}` nesting that we currently have - if there's a flat structure then there is one obvious way to include more info; if it's nested then there's less constraint and it's more difficult to rationalise in future. 

That said, a couple of top-level keys won't kill anyone. I've already built on top of this PR so I'm going to merge into the feature branch and make changes downstream. `request`/`ide` works for me.